src/cacheClient.ts

```diff
@@ -0,0 +1,35 @@
+import NodeCache from '@cacheable/node-cache';
+import { GenerateCacheKeyFn, HyphenEvaluationContext, HyphenProviderOptions } from './types';
+import { cache } from './config';
+
+export class CacheClient {
+  private readonly cache: NodeCache;
+  private readonly generateCacheKeyFn: GenerateCacheKeyFn;
+
+  constructor(cacheConfig?: HyphenProviderOptions['cache']) {
+    const ttlSeconds = cacheConfig?.ttlSeconds ?? cache.ttlSeconds;
```

Comment:
this doesn't work since 0 is a valid ttl

Reply:
@osbornm, should we interpret `0 ` as “use the default value” or as “no expiration”?